ui/src/components/connections/Connections.tsx

```diff
@@ -0,0 +1,179 @@
+          {[...credentials]
+            .sort((a, b) => a.attributes.type.localeCompare(b.attributes.type))
+            .map((credential) => credential.attributes.type)
+            .join(", ")}
+        </Typography.Text>
+      ),
+      title: "Issued credentials",
+    },
+    {
+      dataIndex: "active",
+      key: "active",
+      render: () => <ConnectionsRowDropdown />,
+      width: 55,
+    },
+  ];
+
+  const fetchConnections = useCallback(
+    async (signal: AbortSignal) => {
+      setConnections({ status: "loading" });
+      const response = await getConnections({
+        credentials: true,
+        env,
+        params: {
+          query: queryParam || undefined,
+        },
+        signal,
+      });
+      if (response.isSuccessful) {
+        setConnections({ data: response.data, status: "successful" });
+      } else {
+        if (!isAbortedError(response.error)) {
+          setConnections({ error: response.error, status: "failed" });
+        }
+      }
+    },
+    [env, queryParam]
+  );
+
+  const onSearch = useCallback(
+    (query: string) => {
+      setSearchParams((oldParams) => {
+        const oldQuery = oldParams.get(QUERY_SEARCH_PARAM);
+        const params = new URLSearchParams(oldParams);
+
+        if (query === "") {
+          params.delete(QUERY_SEARCH_PARAM);
+
+          return params;
+        } else if (oldQuery !== query) {
+          params.set(QUERY_SEARCH_PARAM, query);
+
+          return params;
+        }
+        return params;
+      });
+    },
+    [setSearchParams]
+  );
+
+  useEffect(() => {
+    const { aborter } = makeRequestAbortable(fetchConnections);
+
+    return aborter;
+  }, [fetchConnections]);
+
+  const connectionsList = isAsyncTaskDataAvailable(connections) ? connections.data : [];
+
+  return (
+    <SiderLayoutContent
+      description="Connections are established via a secure channel upon issuing credentials to users."
+      title={CONNECTIONS}
+    >
+      <Divider />
+
+      <TableCard
+        defaultContents={
+          <>
+            <Avatar className="avatar-color-cyan" icon={<IconUsers />} size={48} />
+
+            <Typography.Text strong>No connections</Typography.Text>
+
+            <Typography.Text type="secondary">
+              Your connections will be listed here.
+            </Typography.Text>
+          </>
+        }
+        isLoading={isAsyncTaskStarting(connections)}
+        onSearch={onSearch}
+        query={queryParam}
+        searchPlaceholder="Search connections, credentials..."
+        showDefaultContents={
+          connections.status === "successful" && connectionsList.length === 0 && queryParam === null
+        }
+        table={
+          <Table
+            columns={tableContents.map(({ title, ...column }) => ({
+              title: (
+                <Typography.Text type="secondary">
+                  <>{title}</>
+                </Typography.Text>
+              ),
+              ...column,
+            }))}
```

Comment:
Why do we map over the columns to make the title a JSX instead of declaring it as JSX in the first place?

```diff
diff --git a/ui/src/components/connections/Connections.tsx b/ui/src/components/connections/Connections.tsx
index fce23bb..76b1e1e 100644
--- a/ui/src/components/connections/Connections.tsx
+++ b/ui/src/components/connections/Connections.tsx
@@ -38,7 +38,7 @@ export function Connections() {
         </Tooltip>
       ),
       sorter: ({ id: a }, { id: b }) => a.localeCompare(b),
-      title: "Identifier",
+      title: <Typography.Text type="secondary">Identifier</Typography.Text>,
     },
     {
       dataIndex: "credentials",
@@ -52,7 +52,7 @@ export function Connections() {
             .join(", ")}
         </Typography.Text>
       ),
-      title: "Issued credentials",
+      title: <Typography.Text type="secondary">Issued credentials</Typography.Text>,
     },
     {
       dataIndex: "active",
@@ -141,14 +141,7 @@ export function Connections() {
         }
         table={
           <Table
-            columns={tableContents.map(({ title, ...column }) => ({
-              title: (
-                <Typography.Text type="secondary">
-                  <>{title}</>
-                </Typography.Text>
-              ),
-              ...column,
-            }))}
+            columns={tableContents}
             dataSource={connectionsList}
             locale={{
               emptyText:
```

Reply:
I think that should be to not repeat the JSX for each  title

Reply:
It's highly debatable whether doing this map is better than repeating a `Typography.Text` per column.

Reply:
This is reused everywhere, let me know your preferences and the agreement and I'll change it

Reply:
I see, we do it everywhere. Ok, closing this.